fhir-bucket/src/main/java/com/ibm/fhir/bucket/scanner/ImmediateLocalFileReader.java

```diff
@@ -0,0 +1,562 @@
+ */
+package com.ibm.fhir.bucket.scanner;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.StringReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Consumer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.ibm.fhir.bucket.api.BucketLoaderJob;
+import com.ibm.fhir.bucket.api.FileType;
+import com.ibm.fhir.bucket.api.ResourceBundleError;
+import com.ibm.fhir.bucket.api.ResourceEntry;
+import com.ibm.fhir.database.utils.thread.ThreadHandler;
+import com.ibm.fhir.exception.FHIROperationException;
+import com.ibm.fhir.model.format.Format;
+import com.ibm.fhir.model.parser.FHIRParser;
+import com.ibm.fhir.model.parser.exception.FHIRParserException;
+import com.ibm.fhir.model.resource.Bundle;
+import com.ibm.fhir.model.resource.OperationOutcome;
+import com.ibm.fhir.model.resource.Resource;
+import com.ibm.fhir.model.util.FHIRUtil;
+import com.ibm.fhir.validation.FHIRValidator;
+import com.ibm.fhir.validation.exception.FHIRValidationException;
+
+/**
+ * Reads local files directly and processes them to the {@link ResourceHandler} for processing.
+ * Does not rely on COS or the fhirbucket database. Useful for simpler load testing and
+ * loading something like a development database with test data
+ */
+public class ImmediateLocalFileReader {
+    private static final Logger logger = Logger.getLogger(ImmediateLocalFileReader.class.getName());
+
+    // The types of file we are supposed to process
+    private final Set<FileType> fileTypes;
+
+    // The base directory we scan for content to load
+    private final String baseDirectory;
+
+    // The handler which processes resources we've read from COS
+    private final Consumer<ResourceEntry> resourceHandler;
+
+    // The pool used to parallelize loading
+    private final ExecutorService pool;
+
+    // active object thread
+    private Thread mainLoopThread;
+
+    // flow control
+    private Lock lock = new ReentrantLock();
+    private Condition resourceLimit = lock.newCondition();
+    private int inflight = 0;
+
+    // never allow more than this number of items to be allocated to this instance
+    private int maxInflight;
+
+    // ask the database to allocate more items when the current inflight drops below this threshold
```

Comment:
i thought this impl didn't use db?

Reply:
it doesn't - just some copy-paste cruft. I've reworded.